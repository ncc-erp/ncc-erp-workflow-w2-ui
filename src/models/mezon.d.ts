export interface MezonWebView {
  postEvent: (
    eventType: string,
    /* eslint-disable  @typescript-eslint/no-explicit-any */
    eventData: any,
    /* eslint-disable  @typescript-eslint/no-explicit-any */
    callback?: (error?: any) => void
  ) => void;
  /* eslint-disable  @typescript-eslint/no-explicit-any */
  receiveEvent: (eventType: string, eventData: any) => void;
  onEvent: (
    eventType: string,
    /* eslint-disable  @typescript-eslint/no-explicit-any */
    callback: (eventType: string, eventData: any) => void
  ) => void;
  offEvent: (
    eventType: string,
    /* eslint-disable  @typescript-eslint/no-explicit-any */
    callback: (eventType: string, eventData: any) => void
  ) => void;
}

export interface Mezon {
  WebView: MezonWebView;
}

export declare global {
  interface Window {
    Mezon: Mezon;
  }
}
