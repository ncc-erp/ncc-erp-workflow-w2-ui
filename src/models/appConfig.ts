export interface AppConfig {
  openSideBar: boolean;
}
