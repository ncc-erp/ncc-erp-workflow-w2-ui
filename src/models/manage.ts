export interface IPostAndWFH {
  userId: number | undefined;
  userName?: string | undefined;
  numOfPosts: number;
  numOfRequestWFH: number;
  index?: number;
  email?: string;
}

export interface IRender {
  rendered: string;
  protected?: boolean;
}

export interface IHref {
  href: string;
  embeddable?: boolean;
  count?: number;
  id?: number;
  taxonomy?: string;
  name?: string;
  templated?: boolean;
}

export interface IPostANT {
  userName?: string;
  id: number;
  date: string;
  date_gmt: string;
  guid: IRender;
  modified: string;
  modified_gmt: string;
  slug: string;
  status: string;
  type: string;
  link: string;
  title: IRender;
  content: IRender;
  excerpt: IRender;
  author: number;
  featured_media: number;
  comment_status: string;
  ping_status: string;
  sticky: boolean;
  template: string;
  format: string;
  meta: {
    _mi_skip_tracking: false;
    _monsterinsights_sitenote_active: false;
    _monsterinsights_sitenote_note: '';
    _monsterinsights_sitenote_category: 0;
    footnotes: '';
  };
  categories: Array<number>;
  tags: Array<unknown>;
  aioseo_notices: Array<unknown>;
  _links: {
    self: Array<IHref>;
    collection: Array<IHref>;
    about: Array<IHref>;
    author: Array<IHref>;
    replies: Array<IHref>;
    'version-history': Array<IHref>;
    'predecessor-version': Array<IHref>;
    'wp:featuredmedia': Array<IHref>;
    'wp:attachment': Array<IHref>;
    'wp:term': Array<IHref>;
    curies: Array<IHref>;
  };
}

export interface IUserANT {
  id: number;
  name: string;
  url: string;
  description: string;
  link: string;
  slug: string;
  avatar_urls: unknown;
  meta: unknown[];
  simple_local_avatar: unknown;
  _links: {
    self: Array<IHref>;
    collection: Array<IHref>;
  };
}

export interface IFilterReportWFH {
  search: string;
  pages: number;
  per_page: number;
}
