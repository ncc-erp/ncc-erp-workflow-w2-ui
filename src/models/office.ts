

export interface IOffices {
    "displayName": string,
    "code": string,
    "headOfOfficeEmail": string
}

export const officeList: IOffices[] = []