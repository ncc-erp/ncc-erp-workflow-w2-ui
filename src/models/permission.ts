export interface IPermissionItem {
  name: string;
  displayName: string;
  parentName: string;
  isGranted: boolean;
  allowedProviders: string[];
  grantedProviders: [
    {
      providerName: string;
      providerKey: string;
    },
  ];
}
export interface IPermissionList {
  name: string;
  displayName: string;
  permissions: IPermissionItem[];
}

export interface ListResult<T> {
  entityDisplayName: string;
  groups: T[];
}

export interface IUpdateUserPermissionRequest {
  permissions: IPermissionRequestItem[];
}

export interface IPermissionRequestItem {
  name: string;
  isGranted: boolean;
}

export type PermissionResult = ListResult<IPermissionList>;
