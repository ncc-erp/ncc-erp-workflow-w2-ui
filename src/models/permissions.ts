export interface Permissions {
  id: string;
  name: string;
  code: string;
  creationTime: string;
  children?: Permissions[];
}
