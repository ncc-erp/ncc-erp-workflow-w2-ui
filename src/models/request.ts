import { RequestStatus } from 'common/enums';

export interface Request {
  id: string;
  workflowDefinitionId: string;
  workflowDefinitionDisplayName: string;
  settings?: string;
  userRequestName: string;
  createdAt: string;
  lastExecutedAt: string;
  shortTitle?: string;
  status: RequestStatus;
  stakeHolders: string[];
  currentStates: string[];
  creatorId?: string;
}

export interface RequestTemplate {
  id: string;
  displayName: string;
}

export interface FilterRequestParams {
  Status?: string;
  WorkflowDefinitionId?: string;
  maxResultCount: number;
  skipCount: number;
  sorting: string;
  RequestUser?: string;
  StakeHolder?: string;
}

export interface ListResult<T> {
  totalCount: number;
  items: T[];
}

export type FilterRequestResult = ListResult<Request>;

export type RequestTemplateResult = ListResult<RequestTemplate>;

export interface RequestTemplate {
  definitionId: string;
  name: string;
  displayName: string;
  description?: string;
  version: number;
  isSingleton: boolean;
  isPublished: boolean;
  isLatest: boolean;
  inputDefinition: InputDefinition;
  id: string;
}

export interface InputDefinition {
  workflowDefinitionId: string;
  propertyDefinitions: PropertyDefinition[];
  settings?: Settings;
  id: string;
  nameRequest?: string;
}

export interface Settings {
  color: string;
  titleTemplate: string;
}

export interface IJsonObject {
  settings: Settings;
  propertyDefinitions: PropertyDefinition[];
}

export interface PropertyDefinition {
  name: string;
  type: string;
  isRequired: boolean;
  isTitle?: boolean;
  titleTemplate?: string;
}

export interface CreateWorkflowPropertyDefinition {
  name: string;
  isRequired: boolean;
}

export interface ChangeOfficeRequestFormParams {
  currentOffice: string;
  destinationOffice: string;
  content: string;
  startDate: Date;
  endDate: Date;
}

export interface OfficeEquipmentRequestFormParams {
  currentOffice: string;
  reason: string;
}

export interface WfhRequestFormParams {
  currentOffice: string;
  currentProject: string;
  reason: string;
  date: Date;
}

export interface DeviceRequestFormParams {
  currentOffice: string;
  currentProject: string;
  device: string;
  reason: string;
}

export interface IRequestFormParams {
  workflowDefinitionId?: string;
  input: object;
}

export interface ICreateFormParams {
  name: string;
  displayName: string;
  tag: string;
}

export interface IUpdateInputFormParams {
  id: string;
  workflowDefinitionId?: string;
  propertyDefinitions: PropertyDefinition[];
  settings?: Settings;
}

export interface IRequestUser {
  email: string;
  name: string;
  branchName: string;
}

export interface IInputRequest {
  Request: Record<string, Record<string, string> | string>;
  RequestUser: IRequestUser;
}

export interface IOtherActionSignals {
  otherActionSignal: string;
  status: number;
}

export interface ITask {
  name: string;
  email: string;
  emailTo: string[];
  reason?: string;
  status: number;
  workflowInstanceId: string;
  dynamicActionData?: string;
  creationTime: string;
  id: string;
  otherActionSignals?: IOtherActionSignals[];
  description?: string;
  authorName?: string;
  updatedBy?: string;
}

export interface IRequestResult {
  typeRequest: string;
  input: IInputRequest;
  tasks: ITask[];
  workInstanceId: string;
}
