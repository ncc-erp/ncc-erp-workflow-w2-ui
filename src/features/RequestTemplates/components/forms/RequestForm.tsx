import { Button, Text, VStack } from '@chakra-ui/react';
import {
  useOffices,
  useUserProjects,
  useUserInfoWithBranch,
  useNewRequestWorkflow,
  useUserCurrentProject,
} from 'api/apiHooks/requestHooks';
import { SelectField } from 'common/components/SelectField';
import { TextareaField } from 'common/components/TextareaField';
import { TextField } from 'common/components/TextField';
import { useForm } from 'react-hook-form';
import MultiDatePicker, { DateObject } from 'react-multi-date-picker';
import Toolbar from 'react-multi-date-picker/plugins/toolbar';

import DatePicker from 'react-datepicker';
import 'react-datepicker/dist/react-datepicker.css';
import './style.css';

import {
  InputDefinition,
  PropertyDefinition,
  IRequestFormParams,
} from 'models/request';
import { IOffices } from 'models/office';
import { IProjects } from 'models/project';
import { ChangeEvent, useState } from 'react';
import { format } from 'date-fns/esm';
import { useCurrentUser } from 'hooks/useCurrentUser';
import { toast } from 'common/components/StandaloneToast';

interface RequestFormProps {
  inputDefinition?: InputDefinition;
}
type FormParams = Record<
  string,
  string | DateObject | DateObject[] | null | Date | undefined
>;

type FormParamsValue =
  | string
  | DateObject
  | DateObject[]
  | null
  | Date
  | undefined;

const RequestForm = ({ inputDefinition }: RequestFormProps) => {
  const { data: offices, isLoading: isLoading } = useOffices();
  const { data: projects } = useUserProjects();
  const currentUser = useCurrentUser();
  const { data: userInfo } = useUserInfoWithBranch(currentUser.email);
  const { data: userCurrentProject } = useUserCurrentProject();

  const [formParams, setFormParams] = useState<FormParams>({});

  const { handleSubmit } = useForm<FormParams>({});
  const { mutateAsync: createMutate } = useNewRequestWorkflow();
  const formatDate = (date: FormParamsValue) => {
    if (date instanceof Date) {
      return format(date, 'yyyy/MM/dd');
    } else {
      return date?.toString();
    }
  };

  const onSubmit = async () => {
    const formParamsFormatted = { ...formParams };
    Object.keys(formParamsFormatted).forEach((key) => {
      if (
        formParamsFormatted[key] instanceof Date ||
        formParamsFormatted[key] instanceof DateObject ||
        Array.isArray(formParams[key])
      ) {
        formParamsFormatted[key] = formatDate(formParamsFormatted[key]);
      }
    });

    const RequestFormParams: IRequestFormParams = {
      workflowDefinitionId: inputDefinition?.workflowDefinitionId,
      input: formParamsFormatted,
    };

    const { result } = await createMutate(RequestFormParams);
    console.log(result);
    toast({
      description: 'Create Request Successfully',
      status: 'success',
    });
  };

  const handleChangeValue = (
    e:
      | ChangeEvent<HTMLSelectElement | HTMLInputElement | HTMLTextAreaElement>
      | Date
      | DateObject
      | DateObject[]
      | null,
    variable: string
  ) => {
    const updatedFormParams = { ...formParams };
    if (e && 'target' in e) {
      updatedFormParams[variable] = e.target.value;
    } else if (
      Array.isArray(e) ||
      e instanceof Date ||
      e instanceof DateObject
    ) {
      updatedFormParams[variable] = e;
    }
    setFormParams(updatedFormParams);
  };

  const getOptions = (type: string) => {
    switch (type) {
      case 'OfficeList':
        return offices?.map((office: IOffices) => ({
          value: office?.code,
          label: office?.displayName,
        }));

      case 'MyProject':
        return projects?.map((project: IProjects) => ({
          value: project?.code,
          label: project?.name,
        }));
    }
  };

  const getDefaultValueSelected = (type: string, fieldname: string) => {
    switch (type) {
      case 'OfficeList':
        return formParams[fieldname] ?? userInfo?.branch;
      case 'MyProject':
        return formParams[fieldname] ?? userCurrentProject?.code;
    }
  };

  const getField = (Field: PropertyDefinition) => {
    const fieldname = Field?.name ? Field.name : '';
    switch (Field?.type) {
      case 'OfficeList':
      case 'MyProject':
        formParams[fieldname] = getDefaultValueSelected(Field?.type, fieldname);
        return (
          <>
            <Text whiteSpace="nowrap" fontSize="md">
              {fieldname}
              {Field?.isRequired ? (
                <span style={{ color: 'red' }}> *</span>
              ) : (
                ''
              )}
            </Text>
            <SelectField
              name={fieldname}
              size="sm"
              rounded="md"
              options={getOptions(Field?.type) ?? [{ value: '', label: '' }]}
              value={formParams[fieldname] as string}
              onChange={(newValue) => handleChangeValue(newValue, fieldname)}
            />
          </>
        );
      case 'Text':
        formParams[fieldname] = formParams[fieldname] ?? '';
        return (
          <>
            <Text whiteSpace="nowrap" fontSize="md">
              {fieldname}
              {Field?.isRequired ? (
                <span style={{ color: 'red' }}> *</span>
              ) : (
                ''
              )}
            </Text>
            <TextField
              name={fieldname}
              h="50px"
              placeholder={fieldname}
              fontSize="sm"
              onChange={(newValue) => handleChangeValue(newValue, fieldname)}
              value={formParams[fieldname] as string}
            />
          </>
        );
      case 'RichText':
        formParams[fieldname] = formParams[fieldname] ?? '';
        return (
          <>
            <Text whiteSpace="nowrap" fontSize="md">
              {fieldname}
              {Field?.isRequired ? (
                <span style={{ color: 'red' }}> *</span>
              ) : (
                ''
              )}
            </Text>
            <TextareaField
              name={fieldname}
              onChange={(newValue) => handleChangeValue(newValue, fieldname)}
              value={formParams[fieldname] as string}
            />
          </>
        );
      case 'DateTime':
        formParams[fieldname] = formParams[fieldname] ?? new Date();
        if (fieldname != 'Dates')
          return (
            <>
              <Text whiteSpace="nowrap" fontSize="md">
                {Field?.name}
                {Field?.isRequired ? (
                  <span style={{ color: 'red' }}> *</span>
                ) : (
                  ''
                )}
              </Text>
              <DatePicker
                name={fieldname}
                className="datePicker"
                selected={formParams[fieldname] as Date}
                onChange={(newDate) => handleChangeValue(newDate, fieldname)}
                dateFormat="dd-MM-yyyy"
                value={formParams[fieldname] as string}
              />
            </>
          );
        else
          return (
            <>
              <Text whiteSpace="nowrap" fontSize="md">
                {Field?.name}
                {Field?.isRequired ? (
                  <span style={{ color: 'red' }}> *</span>
                ) : (
                  ''
                )}
              </Text>
              <MultiDatePicker
                name={fieldname}
                multiple
                value={formParams[fieldname]}
                onChange={(newDates) => handleChangeValue(newDates, fieldname)}
                plugins={[<Toolbar position="bottom" sort={['close']} />]}
                style={{
                  width: '100%',
                  height: '32px',
                  borderRadius: '5px',
                  fontSize: '16px',
                  padding: '3px 10px',
                }}
              />
            </>
          );
    }
  };

  const renderFormContent = (Fields: PropertyDefinition[] | undefined) => {
    return Fields?.map(function (Field: PropertyDefinition) {
      return getField(Field);
    });
  };

  return (
    <form
      style={{ width: '100%', marginBottom: '20px' }}
      onSubmit={handleSubmit(onSubmit)}
    >
      <VStack spacing="14px" alignItems="flex-start">
        {renderFormContent(inputDefinition?.propertyDefinitions)}
        <Button
          mt="14px"
          h="50px"
          type="submit"
          isLoading={isLoading}
          colorScheme="blackButton"
          w="full"
          textColor="white"
        >
          Save
        </Button>
      </VStack>
    </form>
  );
};

export default RequestForm;
