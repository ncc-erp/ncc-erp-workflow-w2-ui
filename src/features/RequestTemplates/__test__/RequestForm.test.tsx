import { render, screen, waitFor } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import RequestForm from '../components/forms/RequestForm';
import { InputDefinition } from 'models/request';
import userEvent from '@testing-library/user-event';

jest.mock('../../../api/apiHooks/index', () => ({
  useAxios: jest.fn(),
}));

jest.mock('../../../api/axiosInstant', () => ({
  VITE_API_BASE_URL: '/api',
}));

jest.mock('api/apiHooks/requestHooks', () => ({
  useOffices: jest.fn().mockReturnValue({
    data: [
      {
        displayName: 'Đà Nẵng',
        code: 'ĐN',
        headOfOfficeEmail: 'thien.dang@ncc.asia',
      },
      {
        displayName: 'Vinh',
        code: 'V',
        headOfOfficeEmail: 'dai.trinhduc@ncc.asia',
      },
    ],
  }),
  useUserProjects: jest.fn().mockReturnValue({
    data: [
      {
        pm: [
          {
            fullName: 'Khanh Đoàn Công',
            emailAddress: 'khanh.doancong@ncc.asia',
          },
        ],
        code: 'dn-training-php-khanh',
        name: '[ĐN] Training PHP - Khanh',
      },
      {
        pm: [
          {
            fullName: 'Thiên Đặng An',
            emailAddress: 'thien.dang@ncc.asia',
          },
        ],
        code: 'asm',
        name: 'AMS',
      },
      {
        pm: [
          {
            fullName: 'Vân Nguyễn Hồng',
            emailAddress: 'van.nguyenhong@ncc.asia',
          },
          {
            fullName: 'Nhung Nguyễn Thị',
            emailAddress: 'nhung.nguyenthi@ncc.asia',
          },
          {
            fullName: 'Hiền Ngô Thu',
            emailAddress: 'hien.ngothu@ncc.asia',
          },
          {
            fullName: 'Anh Nguyễn Thị Phương',
            emailAddress: 'phuonganh.nguyen@ncc.asia',
          },
        ],
        code: 'nca',
        name: 'Company Activities',
      },
      {
        pm: [
          {
            fullName: 'Tiến Nguyễn Hữu',
            emailAddress: 'tien.nguyenhuu@ncc.asia',
          },
          {
            fullName: 'Trung Đỗ Đức',
            emailAddress: 'trung.doduc@ncc.asia',
          },
          {
            fullName: 'Hiếu Đỗ Hoàng',
            emailAddress: 'hieu.dohoang@ncc.asia',
          },
        ],
        code: 'support',
        name: 'Support',
      },
    ],
  }),
  useUserList: jest.fn().mockReturnValue({
    data: [
      {
        name: 'admin admin',
        email: 'admin@aspnetboilerplate.com',
      },
      {
        name: 'admin admin',
        email: 'admintytyyt@aspnetboilerplate.com',
      },
      {
        name: 'Ân Bùi Hoàng',
        email: 'an.buihoang@ncc.asia',
      },
    ],
  }),
  useUserInfoWithBranch: jest.fn().mockReturnValue({
    data: {
      fullName: 'Huỳnh Bá Nhân',
      email: 'nhan.huynhba@ncc.asia',
      branch: 'ĐN',
    },
  }),
  useUserCurrentProject: jest.fn().mockReturnValue({
    data: {
      pm: {
        fullName: 'Thiên Đặng An',
        emailAddress: 'thien.dang@ncc.asia',
      },
      code: 'asm',
      name: 'AMS',
    },
  }),
  useNewRequestWorkflow: jest.fn(() => ({
    mutateAsync: jest.fn(),
  })),
}));

const inputDefinition: InputDefinition = {
  workflowDefinitionId: '3a05daee-8f55-1e4c-9ac3-443c04309b77',
  propertyDefinitions: [
    {
      name: 'CurrentOffice',
      type: 'OfficeList',
      isRequired: true,
      isTitle: true,
    },
    {
      name: 'DestinationOffice',
      type: 'OfficeList',
      isRequired: true,
      isTitle: true,
    },
    {
      name: 'Content',
      type: 'RichText',
      isRequired: true,
      isTitle: true,
    },
    {
      name: 'StartDate',
      type: 'DateTime',
      isRequired: true,
      isTitle: true,
    },
    {
      name: 'EndDate',
      type: 'DateTime',
      isRequired: false,
      isTitle: true,
    },
  ],
  id: '3a05ffba-3830-e0d4-e931-9381c70a3710',
};

describe('Request Template Form Components', () => {
  const queryClient: QueryClient = new QueryClient();

  test('should match snapshot when rendering', () => {
    const { container } = render(
      <QueryClientProvider client={queryClient}>
        <RequestForm
          inputDefinition={inputDefinition}
          onCloseModal={() => jest.fn()}
        />
      </QueryClientProvider>
    );
    expect(container).toMatchSnapshot();
  });

  describe('Request Form with inputDefinition', () => {
    beforeEach(() => {
      render(
        <QueryClientProvider client={queryClient}>
          <RequestForm
            inputDefinition={inputDefinition}
            onCloseModal={() => jest.fn()}
          />
        </QueryClientProvider>
      );
    });

    describe('Content Input', () => {
      it('should have a Content input when the form is loaded', () => {
        expect(screen.getByText('Content')).toBeInTheDocument();
      });

      it('should show an error message when no content is entered', async () => {
        const submitButton = screen.getByRole('button', {
          name: 'Save',
        });
        userEvent.click(submitButton);
        const result = await screen.findByText(/Content is Required/i);
        expect(result).toBeInTheDocument();
      });
    });

    describe('Start Date Input', () => {
      it('should have a Start Date input when the form is loaded', () => {
        expect(screen.getByLabelText(/Start Date/i)).toBeInTheDocument();
      });

      it('should show an error message when no Start Date is entered', async () => {
        const submitButton = screen.getByRole('button', {
          name: 'Save',
        });
        userEvent.click(submitButton);
        const result = await screen.findByText(/Start Date is Required/i);
        expect(result).toBeInTheDocument();
      });
    });

    describe('End Date Input', () => {
      it('should have an End Date input when the form is loaded', () => {
        expect(screen.getByLabelText(/End Date/i)).toBeInTheDocument();
      });
    });

    describe('When all options are chosen and submitted', () => {
      it('should submit the form with the selected values', async () => {
        userEvent.type(screen.getByText('Content'), 'Is the Content');
        // Select the start date
        const startPicker = screen.getByLabelText(/Start Date/i);
        userEvent.click(startPicker);
        const startDateToSelect = new Date(2023, 9, 18);
        const startDay = startDateToSelect.getDate();

        // Wait for the start date to appear and select it
        await waitFor(() =>
          userEvent.click(screen.getByText(startDay.toString()))
        );

        // Select the end date
        const endPicker = screen.getByLabelText(/End Date/i);
        userEvent.click(endPicker);

        // Wait for the end date to appear and select it
        await waitFor(() => {
          screen.getByText(startDay.toString());
        });
        userEvent.click(screen.getByText(startDay.toString()));

        // Click the "submit" button
        const submitButton = screen.getByRole('button', {
          name: 'Save',
        });
        userEvent.click(submitButton);
      });
    });
  });
});
