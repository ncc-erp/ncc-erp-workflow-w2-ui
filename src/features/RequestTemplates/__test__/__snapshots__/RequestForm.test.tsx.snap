// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request Template Form Components should match snapshot when rendering 1`] = `
<div>
  <form
    style="width: 100%; margin-bottom: 20px;"
  >
    <div
      class="chakra-stack css-z8oacn"
    >
      <div
        class="chakra-form-control css-dynqen"
        role="group"
      >
        <label
          class="chakra-form__label css-8sede8"
          for="field-:r0:"
          id="field-:r0:-label"
        >
          Current Office
          <span
            class="chakra-form__helper-text css-1jd2nkm"
            id="field-:r0:-helptext"
            style="color: red;"
          >
             
            *
          </span>
        </label>
        <div
          class=" css-b62m3t-container"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-2-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class=" css-13cymwt-control"
          >
            <div
              class=" css-1fdsijx-ValueContainer"
            >
              <div
                class=" css-1dimb5e-singleValue"
              >
                Đà Nẵng
              </div>
              <div
                class=" css-qbdosj-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-expanded="false"
                  aria-haspopup="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class=""
                  id="react-select-2-input"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class=" css-1u9des2-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class=" css-1xc3v61-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <input
            name="CurrentOffice"
            type="hidden"
            value="ĐN"
          />
        </div>
      </div>
      <div
        class="chakra-form-control css-dynqen"
        role="group"
      >
        <label
          class="chakra-form__label css-8sede8"
          for="field-:r1:"
          id="field-:r1:-label"
        >
          Destination Office
          <span
            class="chakra-form__helper-text css-1jd2nkm"
            id="field-:r1:-helptext"
            style="color: red;"
          >
             
            *
          </span>
        </label>
        <div
          class=" css-b62m3t-container"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-3-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class=" css-13cymwt-control"
          >
            <div
              class=" css-1fdsijx-ValueContainer"
            >
              <div
                class=" css-1dimb5e-singleValue"
              >
                Đà Nẵng
              </div>
              <div
                class=" css-qbdosj-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-expanded="false"
                  aria-haspopup="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class=""
                  id="react-select-3-input"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class=" css-1u9des2-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class=" css-1xc3v61-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <input
            name="DestinationOffice"
            type="hidden"
            value="ĐN"
          />
        </div>
      </div>
      <div
        class="chakra-form-control css-0"
        role="group"
      >
        <label
          class="chakra-form__label css-c5n4e2"
          for="field-:r2:"
          id="field-:r2:-label"
        >
          Content
          <span
            class="chakra-form__helper-text css-1jd2nkm"
            id="field-:r2:-helptext"
            style="color: red;"
          >
             
            *
          </span>
        </label>
        <div
          class="chakra-form-control css-0"
          role="group"
        >
          <textarea
            class="chakra-textarea css-0"
            id="field-:r3:"
            name="Content"
            placeholder="Content"
          />
        </div>
      </div>
      <div
        class="chakra-form-control css-0"
        role="group"
      >
        <label
          class="chakra-form__label css-c5n4e2"
          for="StartDate"
          id="field-:r4:-label"
        >
          Start Date
          <span
            class="chakra-form__helper-text css-1jd2nkm"
            id="field-:r4:-helptext"
            style="color: red;"
          >
             
            *
          </span>
        </label>
        <div
          class="react-datepicker-wrapper"
        >
          <div
            class="react-datepicker__input-container"
          >
            <input
              autocomplete="off"
              class=""
              id="StartDate"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="chakra-form-control css-0"
        role="group"
      >
        <label
          class="chakra-form__label css-c5n4e2"
          for="EndDate"
          id="field-:r5:-label"
        >
          End Date
        </label>
        <div
          class="react-datepicker-wrapper"
        >
          <div
            class="react-datepicker__input-container"
          >
            <input
              autocomplete="off"
              class=""
              id="EndDate"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <button
        class="chakra-button css-1jntfhr"
        type="submit"
      >
        Save
      </button>
    </div>
  </form>
</div>
`;
